### ISSUE TITLE
A column block defined in a post type template with a unitless width gets a flex-basis in px instead of %

### ISSUE THREAD

**Issue by coreymckrill** [WordPress/gutenberg#26641]
**Describe the bug**
When a post type template includes a definition for a Columns block, and an individual Column block within it specifies a width attribute as a unitless number, that number is now interpreted as pixels, whereas in the past it was interpreted as a percentage. This is a breaking change for existing post type templates.

**To reproduce**
1. Register a new post type and give it a template like:
    ```
    'template' => [
        [
            'core/columns',
            [
                [ 'core/column', [ 'width' => 66.66 ] ]
            ],
            [
                [ 'core/column', [ 'width' => 33.33 ] ]
            ], 
        ],
    ]
    ```
1. Navigate to the Add New screen for that post type.
1. The columns will be very narrow. Switching to code view, you can see that the block metadata has `wp:column {"width":66.66}`, but then the style attribute added to the column div is `style="flex-basis:66.66px"`.

**Expected behavior**
Until recently, a unitless number for the width was interpreted as a percentage instead of absolute pixels. So I was expecting the style attribute to be `style="flex-basis:66.66%"`. I'm not sure which commit caused the change, but it might be [this one](https://github.com/WordPress/gutenberg/commit/1cad0e16c1fd8e46a57c861e69a86a4d3ee8bbd8).

**Screenshots**
https://d.pr/i/G4vgpk

**Editor version:**
- WordPress version: **5.6-alpha-49226**
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? **default**

**Desktop:**
 - OS: **MacOS**
 - Browser: **Firefox**
 - Version **82**

**Comment by tellthemachines** [comment]
Hmm, would this be related to #26096 or are they separate issues?

**Comment by coreymckrill** [comment]
Definitely could be related to the same change, but it seems like they are dealing with separate effects.

**Comment by gziolo** [comment]
@tellthemachines, do you work on this issue?

**Comment by tellthemachines** [comment]
@gziolo yup, should have a PR up soon!

**Comment by tellthemachines** [comment]
Closing as this has been fixed by #26757 .
